docs/docs/classic-ml/deep-learning/pytorch/guide/index.mdx

```diff
@@ -0,0 +1,665 @@
+    test_loss = test_loss / len(test_loader)
+    test_acc = 100.0 * test_correct / test_total
+
+    mlflow.log_metrics({"test_loss": test_loss, "test_accuracy": test_acc})
+
+    print(f"Final Test Accuracy: {test_acc:.2f}%")
+```
+
+If you run this code with a local MLflow server, you'll see comprehensive tracking in the MLflow UI:
+
+<div className="center-div" style={{ width: "90%" }}>
+  ![Basic Example of PyTorch with MLflow](/images/deep-learning/pytorch/guide/pytorch-guide-basic-example-ui.png)
+</div>
+
+## Saving Your PyTorch Model to MLflow
+
+### Basic Model Saving
+
+MLflow makes it easy to save and load PyTorch models for reproducible inference:
+
+```python
+import mlflow
+import numpy as np
+import torch
+import torch.nn as nn
+
+
+# Define model
+class NeuralNetwork(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.flatten = nn.Flatten()
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(28 * 28, 512),
+            nn.ReLU(),
+            nn.Linear(512, 512),
+            nn.ReLU(),
+            nn.Linear(512, 10),
+        )
+
+    def forward(self, x):
+        x = self.flatten(x)
+        logits = self.linear_relu_stack(x)
+        return logits
+
+
+model = NeuralNetwork()
+
+# Train your model (code omitted for brevity)
+
+with mlflow.start_run() as run:
+    # Log the model
+    mlflow.pytorch.log_model(model, "model")
+
+# Load and use the model
+logged_model = f"runs:/{run.info.run_id}/model"
+loaded_model = mlflow.pyfunc.load_model(logged_model)
+
+# Make predictions
+sample_input = np.random.uniform(size=[1, 28, 28]).astype(np.float32)
+predictions = loaded_model.predict(sample_input)
+print("Predictions:", predictions)
+```
+
+<details>
+  <summary>TorchScript Compatibility</summary>
+
+  MLflow seamlessly works with TorchScript, which can optimize your models for production:
+
+  ```python
+  # Create a scripted version of your model
+  scripted_model = torch.jit.script(model)
+
+  # Log the scripted model to MLflow
+  with mlflow.start_run() as run:
+      mlflow.pytorch.log_model(scripted_model, "scripted_model")
+
+  # The logged model will contain the compiled graph
+  ```
+
+  Benefits of using TorchScript with MLflow:
+  - ⚡ **Performance Optimization**: Compiled graphs for faster inference
+  - 🔒 **Deployment Security**: Protected model architecture for secure deployment
+  - 🌉 **Language Interoperability**: Use models in C++ environments
+  - 📱 **Mobile Deployment**: Optimized for resource-constrained devices
+</details>
+
+You can view the saved model in the MLflow UI:
+
+<div className="center-div" style={{ width: "90%" }}>
+  ![Basic PyTorch Saving](/images/deep-learning/pytorch/guide/pytorch-guide-basic-saving.png)
+</div>
+
+### Model Signatures
+
+A model signature defines the schema of inputs and outputs, enhancing model understanding and enabling validation:
+
+<details>
+  <summary>Automatic Signature Inference</summary>
```

Comment:
Could we swap the order? Putting manual signature here and auto signature inference as the default one, since infer_signature is much easier to use?